### ISSUE TITLE
Error when compiling Realm for Swift 4.2 with Xcode beta 1. 

### ISSUE THREAD

**Issue by charvoa** [realm/realm-cocoa#5820]
## Goals
Compile Realm for Swift 4.2 with Xcode 10 beta 1 

## Expected Results
Compile to go smooth

## Actual Results

Logs : 

error: Build input file cannot be found: '/Users/nicolascharvoz/Documents/Work/***/realm-cocoa/RealmSwift/SwiftVersion.swift'
[...]
The following build commands failed:
CompileSwiftSources normal arm64 com.apple.xcode.tools.swift.compiler
CompileSwiftSources normal armv7 com.apple.xcode.tools.swift.compiler
(2 failures)

## Steps to Reproduce
git clone --recursive https://github.com/realm/realm-cocoa.git
cd realm-cocoa
DEVELOPER_DIR=/Applications/Xcode-beta.app/Contents/Developer sh build.sh ios-swift

## Version of Realm and Tooling
```
ProductName:	Mac OS X
ProductVersion:	10.13.4
BuildVersion:	17E202

/usr/local/bin/pod
1.5.3


/bin/bash
GNU bash, version 3.2.57(1)-release (x86_64-apple-darwin17)

carthage not found
(not in use here)

/usr/bin/git
git version 2.15.1 (Apple Git-101)
```

**Comment by tgoyne** [comment]
That file is automatically created during the build process. I suspect we don't have the dependencies set up for it quite correctly, so the new build system will sometimes decide to generate it too late.

**Comment by tgoyne** [comment]
This should have been fixed by #5891.
